utils/src/server/rpc_module.rs

```diff
@@ -211,6 +228,86 @@ impl<Context> RpcContextModule<Context> {
 		Ok(())
 	}
 
+	/// Register a new RPC subscription that invokes callback on every subscription request.
+	/// The callback itself takes three parameters:
+	/// 	- RpcParams: JSONRPC parameters in the subscription request.
+	/// 	- SubscriptionSink: A sink to send messages to the subscriber.
+	/// 	- Context: Any type that can be embedded into the RpcContextModule.
+	///
+	/// # Examples
+	///
+	/// ```no_run
+	///
+	/// use jsonrpsee_utils::server::rpc_module::RpcContextModule;
+	///
+	/// let mut ctx = RpcContextModule::new(99_usize);
+	/// ctx.register_subscription_with_context("sub", "unsub", |params, sink, ctx| {
+	///		let x: usize = params.one()?;
+	///		std::thread::spawn(move || {
+	///			let sum = x + (*ctx);
+	///			sink.send(&sum)
+	///		});
+	///		Ok(())
+	/// });
+	/// ```
+	pub fn register_subscription_with_context<F>(
+		&mut self,
+		subscribe_method_name: &'static str,
+		unsubscribe_method_name: &'static str,
+		callback: F,
+	) -> Result<(), Error>
+	where
+		Context: Send + Sync + 'static,
+		F: Fn(RpcParams, SubscriptionSink, Arc<Context>) -> Result<(), Error> + Send + Sync + 'static,
```

Comment:
This is an `Arc` because you would most likely to move the `Context` into a task or thread then &Context won't work